starter-code/scripts/articles.js

```diff
@@ -4,12 +4,21 @@ function Article (opts) {
   // TODO: Use the object passed in to complete this constructor function:
   // Save ALL the properties of `opts` into `this`.
   this.author = opts.author;
+  this.authorUrl = opts.authorUrl;
+  this.title = opts.title;
+  this.body = opts.body;
+  this.publishedOn = opts.publishedOn;
+  this.category = opts.category;
 }
 
 Article.prototype.toHtml = function() {
   var $newArticle = $('article.template').clone();
   $newArticle.attr('data-category', this.category);
-
+  $newArticle.find('address a').text(this.author);
+  $newArticle.find('a').attr('href', this.authorUrl);
+  $newArticle.find('header h1').text(this.title);
+  $newArticle.find('.article-body').html(this.body);
+  $newArticle.find('time').before(this.publishedOn);
```

Comment:
We use the .find() method to select the particular html we'd like to update (the link, for instance), and then set using the appropriate updating method -- text is updated with text(), html is placed with html(), and the source for the url is set using an attr() update.

This bit of code successfully selects and sets each item in the template clone using the article constructor for the this's.